oop/src/PurchaseOrders/domain/PurchaseOrder.entity.ts

```diff
@@ -0,0 +1,22 @@
+import { createUuid, UUID } from "../../utilities/uuid";
+import { PurchaseOrderLineItem } from "./PurchaseOrderLineItem.entity";
+
+export class PurchaseOrder {
+  private _id: UUID;
+  private _lineItems: PurchaseOrderLineItem[] = [];
```

Comment:
Consider creating a factory for the PurchaseOrder entity since it is an aggregate (it contains line items). As this becomes more complex the constructor will become overburdened.